core/util/eventtarget.js

```diff
@@ -8,33 +8,28 @@
 
 export default class EventTargetMixin {
     constructor() {
-        this._listeners = null;
+        this._listeners = new Map();
     }
 
     addEventListener(type, callback) {
-        if (!this._listeners) {
-            this._listeners = new Map();
-        }
         if (!this._listeners.has(type)) {
             this._listeners.set(type, new Set());
         }
         this._listeners.get(type).add(callback);
     }
 
     removeEventListener(type, callback) {
-        if (!this._listeners || !this._listeners.has(type)) {
-            return;
+        if (this._listeners.has(type)) {
```

Comment:
Very minor nitpick, but is it worth having symmetry with `dispatchEvent()`?

Reply:
The difference is that dispatch returns a boolean and this doesn't return anything.

That would mean:
```javascript
        if (!this._listeners.has(type)) {
            return;
        }

        this._listeners.get(type).delete(callback);
```

which feels a bit unnecessary to me.

But I can change it if you would prefer it 🙂

Reply:
I like it the way it is, the functions are different enough.